chaos-days/blog/2024-07-25-Using-flow-control-to-handle-bottlenecked-exporting/index.md

```diff
@@ -2,66 +2,60 @@
 date:   2024-07-25
-categories: 
-  - performance 
+categories:
+  - performance
 tags:
   - availability
 authors: rodrigo
 ---
 
-# Using flow control to handle bottleneck on exporting 
+# Using flow control to handle bottleneck on exporting
 
-With the Zeebe now having a unified flow control that protects from user commands that cannot be processed fast enough and from excessive writes, we wanted to test situations when exporting is degraded and how to configure the rate limits for writes to bring the cluster back to a balanced/safe state.
+Zeebe 8.6 introduces a new unified flow control mechanism that is able to limit user commands (by default it tries to achieve 200ms response times) and rate limit writes of new records in general (disabled by default).
+Limiting the write rate is a new feature that can be used to prevent building up an excessive exporting backlog.
+There are two ways to limit the write rate, either by setting a static limit or by enabling throttling that dynamically adjust the write rate based on the exporting backlog and rate.
+In these experiments, we will test both ways of limiting the write rate and observe the effects on processing and exporting.
 
 **TL;DR;**
+Both setting a static write rate limit and enabling throttling of the write rate can be used to prevent building up an excessive exporting backlog.
+For users, this will be seen as backpressure because processing speed is limited by the rate at which it can write processing results.
 
-Enabling the write rate limiting and throttling makes the processing and exporting speed match and from a user's perspective, the reduced speed is seen as backpressure.
+## Static write limit
 
-## Degraded exporting performance should result in backpressure
+We will construct a cluster under normal utilization and then artificially degrade the exporting process.
+After this we will apply flow control settings to statically rate limit all writes.
+The limit will be set slightly lower than the observed exporting rate.
 
-The new unified flow control introduces a mechanism that rate limits all writes based on the exporting rate. If exporting slows down, writing of new records becomes slower too. For user commands, this will show up as increased backpressure. Internal processing will slow down because the stream processor needs to wait longer for the writing of processing results to complete.
-
-To test this, we will construct a cluster under normal utilization and then artificially degrade the exporting process. After this we will configure the the flow control to rate limit all writes to be on par with the exported records.
-
-For this, we have two possibilities we can enable a static write rate limit or enable throttling that keeps the writes bounded by the exporter backlog. For this we will use the unified control endpoint and configure write rate limit.
+For this we will use the flow control endpoint to temporarily configure the write rate limit.
```

Comment:
💭 Should we remove the word "temporarily" here? I think it can induce the reader that the changes are not permanent if we don't configure them back to the original values.

Reply:
Feel free to add more explanation to this but IMO it's nice if we mention that changes done through the actuator are not persisted and temporary.